### ISSUE TITLE
TBD: Access to dividends with ST tokens

### ISSUE THREAD

**Issue by pabloruiz55** [PolymathNetwork/polymath-core-deprecated#70]
If I'm not mistaken, this hasn't been laid out yet, but one of the main capabilities of Security tokens should be giving the issuer the ability to grant access to dividends to their holders.

**Comment by adamdossa** [comment]
Note - the current ST implementation would not allow dividends easily as you can't query balances as of a specific date (e.g. the dividend date).
https://medium.com/@adam.dossa/dividend-payment-contract-based-on-minimetoken-90e5ba7c85d5

**Comment by sangheraio** [comment]
Yes this hasn't been implemented into the spec, as dividends could be done through off chain payments (using idnetities from the KYC provider), or the issuer purchasing a large amount of security tokens and burning.

I think the gas costs of minime token are not worth it when there are alternative methods for dividends. @adamdossa, do you want to work on a proposal for on chain dividends? If we could get something in RC1 I think that would be a really cool feature.

**Comment by adamdossa** [comment]
Yep - I can do this early next week (put together a proposal for a light-weight on-chain dividend mechanism).

**Comment by adamdossa** [comment]
As discussed on Friday's call, for now we will leave dividends as being an off-chain process and look at enhancing ST with this feature on-chain as a V2 / enhancement.

**Comment by pabloruiz55** [comment]
Moved to V2 milestone
